components/scream/src/physics/rrtmgp/atmosphere_radiation.cpp

```diff
@@ -1,52 +1,272 @@
+        m_required_fields.emplace("sfc_alb_dir", scalar2d_swband_layout, nondim, grid->name());
+        m_required_fields.emplace("sfc_alb_dif", scalar2d_swband_layout, nondim, grid->name());
+        m_required_fields.emplace("mu0", scalar2d_layout, nondim, grid->name());
+        m_required_fields.emplace("lwp", scalar3d_layout_mid, kg/m3, grid->name());
+        m_required_fields.emplace("iwp", scalar3d_layout_mid, kg/m3, grid->name());
+        m_required_fields.emplace("rel", scalar3d_layout_mid, micron, grid->name());
+        m_required_fields.emplace("rei", scalar3d_layout_mid, micron, grid->name());
+
+        // Outputs; these needed to be added to both required and computed fields?
+        m_required_fields.emplace("sw_flux_dn", scalar3d_layout_int, Wm2, grid->name());
+        m_required_fields.emplace("sw_flux_up", scalar3d_layout_int, Wm2, grid->name());
+        m_required_fields.emplace("sw_flux_dn_dir", scalar3d_layout_int, Wm2, grid->name());
+        m_required_fields.emplace("lw_flux_up", scalar3d_layout_int, Wm2, grid->name());
+        m_required_fields.emplace("lw_flux_dn", scalar3d_layout_int, Wm2, grid->name());
+
+        // Set computed (output) fields
+        m_computed_fields.emplace("sw_flux_dn", scalar3d_layout_int, Wm2, grid->name());
+        m_computed_fields.emplace("sw_flux_up", scalar3d_layout_int, Wm2, grid->name());
+        m_computed_fields.emplace("sw_flux_dn_dir", scalar3d_layout_int, Wm2, grid->name());
+        m_computed_fields.emplace("lw_flux_up", scalar3d_layout_int, Wm2, grid->name());
+        m_computed_fields.emplace("lw_flux_dn", scalar3d_layout_int, Wm2, grid->name());
+
     }  // RRTMGPRadiation::set_grids
 
     void RRTMGPRadiation::initialize_impl(const util::TimeStamp& t0) {
-        // Call RRTMGP initialize routine
-        //rrtmgp_initialize()
+        // We may have to init some fields from within RRTMGP. This can be the case in a RRTMGP standalone run.
+        // Some options:
+        //  - we can tell RRTMGP it can init all inputs or specify which ones it can init. We call the
+        //    resulting list of inputs the 'initializable' (or initable) inputs. The default is
+        //    that no inputs can be inited.
+        //  - we can request that RRTMGP either inits no inputs or all of the initable ones (as specified
+        //    at the previous point). The default is that RRTMGP must be in charge of init ing ALL or NONE
+        //    of its initable inputs.
+        // Recall that:
+        //  - initable fields may not need initialization (e.g., some other atm proc that
+        //    appears earlier in the atm dag might provide them).
+        std::vector<std::string> rrtmgp_inputs = {
+            "pmid","pint","tmid","tint","col_dry","gas_vmr","sfc_alb_dir","sfc_alb_dif",
+            "mu0","lwp","iwp","rel","rei",
+            "sw_flux_up", "sw_flux_dn", "sw_flux_dn_dir", "lw_flux_up", "lw_flux_dn"
```

Comment:
I see here that you are initializing fields you call "output" above.  Are they actually being used as input/output?  If so, you may want to change the comment on line 72, otherwise future developers may be confused.

Reply:
Good catch. I think I included these when I was thinking I needed to register these with the AD here. Removed.